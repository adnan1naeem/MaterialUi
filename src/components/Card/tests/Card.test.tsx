import { Card, CardBody, CardFooter, CardHeader, CardMedia } from '../';
import renderer from 'react-test-renderer';
import React from 'react';


test('Card component with tag props', () => {
	const card = renderer.create(
		<Card tag="span"></Card>
  ).toJSON();
	expect(card.props.tag).toEqual('span');
	expect(card).toMatchSnapshot();
});

test('Card component with children props', () => {
	const card = renderer.create(
		<Card tag="span"><div>Hello</div></Card>
  ).toJSON();
	expect(card.props.children).toEqual(undefined);
	expect(card).toMatchSnapshot();
});

test('CardMedia with children props', () => {
	const cardmedia = renderer.create(
		<CardMedia className="span" src="https://material-ui-next.com/static/images/remy.jpg"></CardMedia>
  ).toJSON();
	expect(cardmedia.props.className).toEqual('MuiCardMedia-root-28 span');
	expect(cardmedia).toMatchSnapshot();
});

test('CardBody with children props', () => {
	const cardbody= renderer.create(
		<CardBody className="span" style={{ width:200 }}></CardBody>
  ).toJSON();
	expect(cardbody.props.style).toEqual({ width:200 });
	expect(cardbody).toMatchSnapshot();
});

test('CardHeader with children props', () => {
	const cardheader= renderer.create(
		<CardHeader className="span" title="hello">second</CardHeader>
  ).toJSON();
	expect(cardheader.props.title).toEqual(undefined);
	expect(cardheader).toMatchSnapshot();
});

test('CardFooter with children props', () => {
	const cardfooter=renderer.create(
		<CardFooter className="span">second</CardFooter>
  ).toJSON();
	expect(cardfooter.props.children).toEqual(undefined);
	expect(cardfooter).toMatchSnapshot();
});