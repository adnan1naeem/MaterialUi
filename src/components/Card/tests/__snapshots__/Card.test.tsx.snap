// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component with children props 1`] = `
<span
  className="MuiPaper-root-1 MuiPaper-shadow2-5 MuiPaper-rounded-2"
  style={undefined}
  tag="span"
>
  <div>
    Hello
  </div>
</span>
`;

exports[`Card component with tag props 1`] = `
<span
  className="MuiPaper-root-1 MuiPaper-shadow2-5 MuiPaper-rounded-2"
  style={undefined}
  tag="span"
/>
`;

exports[`CardBody with children props 1`] = `
<div
  className="MuiCardContent-root-30 span"
  style={
    Object {
      "width": 200,
    }
  }
/>
`;

exports[`CardFooter with children props 1`] = `
<div
  className="MuiCardActions-root-61 span"
/>
`;

exports[`CardHeader with children props 1`] = `
<div
  className="MuiCardContent-root-30 MuiCardHeader-root-31 span"
  style={undefined}
>
  <div
    className="MuiCardHeader-content-34"
  >
    <span
      className="MuiTypography-root-37 MuiTypography-headline-42 MuiCardHeader-title-35"
    >
      hello
    </span>
    <span
      className="MuiTypography-root-37 MuiTypography-body1-46 MuiTypography-colorSecondary-58 MuiCardHeader-subheader-36"
    >
      second
    </span>
  </div>
</div>
`;

exports[`CardMedia with children props 1`] = `
<div
  className="MuiCardMedia-root-28 span"
  src={undefined}
  style={
    Object {
      "backgroundImage": "url(https://material-ui-next.com/static/images/remy.jpg)",
      "height": 200,
    }
  }
/>
`;
